docs/DeveloperGuide.md

```diff
@@ -241,6 +244,42 @@ _{more aspects and alternatives to be added}_
 _{Explain here how the data archiving feature will be implemented}_
 
 
+### <a name="adding_a_session"></a>Adding a session
+
+Adding a session to a class requires user input from the CLI, then the parser will parse the user input and 
+to obtain the name and the date of the session. The newly added session will also be filled with students' 
+default presence and participation.
+
+The below example given assumes that the user inputs `addses s/Tutorial 1 d/1/1/2020`, which 'Tutorial 1' and
+'1/1/2020' are the name and the date of the session to be added, and `addses` command has already been parsed.
+
+Step 1: Parse input and create session
+
+A `AddSessionCommandParser` is created by `LogicManager` and calls the parse method with the given
+user input. Input is parsed according to the prefix `s/` and `d/` to identify the parts of the 
+user input and split it into String slices. If both the name and date are present in the input, 
+those Strings will be used to create a `SessionName` and a `SessionDate` object, which will be used to 
+create and initialize a `Session` object. The `Session` object created will be the one being added to 
+the current session list later.
+
+The parse function returns a `AddSessionCommand` object that contains the session object
+ back to the `LogicManager`.
+
+![AddSessionSequenceDiagram1](images/AddSessionSequenceDiagram1.png)
+
+Step 2: Add session to the model/session list
+
+`LogicManager` calls execute method to proceed on to adding the session to the model.
+The `AddSessionCommand` will first check if the same session is already in the session list, if the 
+session exists in the current session list, then the session will not be added to the session list.
+
+Assuming that the session to add is not a session already in the list, `ModelManager` will update
+the internal student list of the session list, then it uses addSession method to add the session to the 
```

Comment:
Perhaps breaking this paragraph into two sentences will make it flow better? For example: "... . It then uses `ModelManager#addSession()` .."